saq/types.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+import typing as t
+
+from saq.compat import Literal, TypedDict
+
+if t.TYPE_CHECKING:
+    from saq.job import Job, Status
+
+BeforeEnqueueType = t.Callable[["Job"], t.Awaitable[t.Any]]
+Context = t.Dict[str, t.Any]
+CountKind = Literal["queued", "active", "incomplete"]
+DumpType = t.Callable[[t.Dict], str]
+DurationKind = Literal["process", "start", "total", "running"]
+Function = t.Callable
```

Comment:
This is just a stub for a better type.

I tried different configurations with `Protocol` and `__call__` and `@overload` to try and satisfy requirement that these can be both sync and async, accept a single pos arg, either accept kwargs, or not, as well as having the `__qualname__` attribute, but couldn't find a solution that worked.

Reply:
Going 3.8+ might help with this as positional only parameters were introduced in 3.8.